timecache/time_cache.go

```diff
@@ -8,31 +8,45 @@ import (
 
 var log = logger.Logger("pubsub/timecache")
 
+// Stategy is the TimeCache expiration strategy to use.
 type Strategy uint8
 
 const (
+	// Strategy_FirstSeen expires an entry from the time it was added.
 	Strategy_FirstSeen Strategy = iota
+	// Stategy_LastSeen expires an entry from the last time it was touched by an Add or Has.
 	Strategy_LastSeen
 )
 
+// TimeCache is a cahe of recently seen messages (by id).
 type TimeCache interface {
-	Add(string)
+	// Add adds an id into the cache, if it is not already there.
+	// Returns true if the id was newly added to the cache.
+	// Depending on the implementation strategy, it may or may not update the expiry of
+	// an existing entry.
+	Add(string) bool
+	// Has checks the cache for the presence of an id.
+	// Depending on the implementation strategy, it may or may not update the expiry of
+	// an existing entry.
 	Has(string) bool
+	// Done signals that the user is done with this cache, which it may stop background threads
+	// and relinquish resources.
+	Done()
```

Comment:
Is `.Close()` not the preferred convention? `.Done()` is used in `context.Context` which returns a channel that's closed when the context is finished. `.Done()` is also used in waitgroups to signal completion (but not cleanup).

`.Close()` seems closer to the intent. 

That said. This is minor, really just a nit

Reply:
close returns error... there is none here, and there is no point.

Reply:
Initially i used Closer actually, but then i realize i am making work for myself with th error ;)